src/main/takeDetection/takeDetection.ts

```diff
@@ -48,56 +48,202 @@ export function findSentences(words: Word[]): Sentence[] {
+  // assume sentences in potential take are similar
+  let isSimilarTake = true;
+  let updatedCurrentSentenceIdx = 0;
+
+  // compare sentences within the potential take length, start from the second sentence
+  for (let i = 1; i < potentialTakeLen; i += 1) {
+    // if sentence is not similar, discard potential take and restart
+    if (!isSimilarTake) {
+      break;
+    }
+
+    // compute the next sentence's index in the first take group
+    // as reference to compare the rest
+    const nextSentenceIdxToCompare = potentialTakeStartIdxs[0] + i;
+
+    // compare sentence in first potential take chunk with every other potential take chunk
+    // compare the next sentence each time
+    for (let j = 1; j < potentialTakeStartIdxs.length; j += 1) {
+      const nextSimilarity = getSimilarityScore(
+        sentences[nextSentenceIdxToCompare].sentenceString,
+        sentences[potentialTakeStartIdxs[j] + i].sentenceString
+      );
+
+      // if sentence in the chunk is not similar, discard all chunks to the right
+      if (nextSimilarity < THRESHOLD) {
+        const potentialTakeStartIdxsLen = potentialTakeStartIdxs.length;
+        const chunksToDiscard =
+          potentialTakeStartIdxsLen - potentialTakeStartIdxs[j];
+        const chunksLeft = potentialTakeStartIdxsLen - chunksToDiscard;
+
+        // if only the first chunk left, reset
+        if (chunksLeft === 1) {
+          // if potential take greater then two, re-start from the first sentence in the second chunk
+          if (potentialTakeStartIdxsLen > 2) {
+            updatedCurrentSentenceIdx = potentialTakeStartIdxs[j] + i - 1;
+          } else {
+            // else start from the failed sentence
+            updatedCurrentSentenceIdx = potentialTakeStartIdxs[j] + i;
+          }
+
+          isSimilarTake = false;
+          break;
+        }
+
+        // discard chunks to the right
+        for (let k = 0; k < chunksToDiscard; k += 1) {
+          potentialTakeStartIdxs.pop();
+        }
+      }
+    }
+  }
+
+  if (isSimilarTake)
+    updatedCurrentSentenceIdx =
+      potentialTakeLen * potentialTakeStartIdxs.length;
+
+  return { isSimilarTake, updatedCurrentSentenceIdx };
+};
+
+const newTakeGroup = (
+  potentialTakeStartIdxs: number[],
+  potentialTakeLen: number,
+  sentences: Sentence[]
+): InjectableTakeGroup => {
+  // if all the sentence in potential take are similar, add them to takeGroups
   const newGroup: InjectableTakeGroup = { takes: [] };
-  let nextTakeRange: IndexRange = {
-    startIndex: sentenceOne.startIndex as number,
-    endIndex: sentenceOne.endIndex as number,
-  };
-  let nextTake: InjectableTake = { wordRange: nextTakeRange };
-  newGroup.takes.push(nextTake);
-  nextTakeRange = {
-    startIndex: sentenceTwo.startIndex as number,
-    endIndex: sentenceTwo.endIndex as number,
-  };
-  nextTake = { wordRange: nextTakeRange };
-  newGroup.takes.push(nextTake);
+
+  potentialTakeStartIdxs.forEach((startIdx) => {
```

Comment:
Since this is unconditionally pushing to an array you can replace the whole thing with a `.map`

i.e. the entire function body would become

```
return potentialTakeStartIdxs.map((startIdx) => {
  const takeStartSentence = sentences[startIdx];
  const takeEndSentence = sentences[startIdx + potentialTakeLen - 1];

  return {
    wordRange: {
      startIndex: takeStartSentence.startIndex as number,
      endIndex: takeEndSentence.endIndex as number,
    },
  } as InjectableTake;
});
```

Also I noticed take length is constant across an entire take group. What happens if two takes are similar but contain a different number of sentences - is that just not allowed? I can see it happening e.g.

take1: "Thanks for watching, everybody. Please don't forget to like and subscribe, and leave a comment below."
take2: "I appreciate every one of you watching this so please hit like and subscribe as well as leaving a comment."

Obviously a contrived example, but you get the point.

Reply:
I think we decided similar takes should have the same length. 

Because currently, we can only compare two sentences each time. It is hard to detect one sentence that has a similar meaning to several sentences combined, unless we compute all combinations of the sentences

Reply:
Okay, it does seem easier

You could do a multi sentence comparison using the geometric mean of all cross-similarities